ui/apps/platform/src/Containers/NetworkGraph/components/TimeWindowSelector.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { Select, SelectOption } from '@patternfly/react-core';
+
+import { timeWindows } from 'constants/timeWindows';
+import useSelectToggle from 'hooks/patternfly/useSelectToggle';
+
+type TimeWindowSelectorProps = {
+    setActiveTimeWindow: (timeWindow) => void;
+    activeTimeWindow: string;
+    isDisabled?: boolean;
+};
+
+function TimeWindowSelector({
+    setActiveTimeWindow,
+    activeTimeWindow,
+    isDisabled = false,
+}: TimeWindowSelectorProps) {
+    const { closeSelect, isOpen, onToggle } = useSelectToggle();
+    function selectTimeWindow(_event, selection) {
+        closeSelect();
+        setActiveTimeWindow(selection);
+    }
+
+    return (
+        <Select
+            isOpen={isOpen}
+            onToggle={onToggle}
+            onSelect={selectTimeWindow}
+            selections={activeTimeWindow}
+            isDisabled={isDisabled}
+        >
+            {timeWindows.map((window) => (
```

Comment:
I know that this is shadowed correctly, but naming a local var the same thing as the global `window` object makes me uneasy.